core/src/OpenACC/Kokkos_OpenACC_Instance.cpp

```diff
@@ -56,7 +60,12 @@ Kokkos::Experimental::Impl::OpenACCInternal::singleton() {
   return &self;
 }
 
-void Kokkos::Experimental::Impl::OpenACCInternal::initialize() {
+void Kokkos::Experimental::Impl::OpenACCInternal::initialize(
+    InitializationSettings const& settings) {
+  using Kokkos::Impl::get_gpu;
+  int dev_num           = get_gpu(settings);
+  acc_device_t dev_type = OpenACC_Traits::dev_type;
+  acc_set_device_num(dev_num, dev_type);
```

Comment:
`acc_device_default` does not distinguish host from accelerator devices, and thus if we want to allow only accelerator devices as an OpenACC backend target, we will have to change the default device type to `acc_device_no_host` and add extra code to check whether the current device is a host or not for the non-default cases.

Reply:
For `SYCL` we accept any kind of device if no device id was specified. If a device id is specified, we error out if the device id is larger than the number of devices.

Reply:
We could do something like
```C++
int num_devices = acc_get_num_devices(
       Kokkos::Experimental::Impl::OpenACC_Traits::dev_type);
if (num_devices == 0 && !settings.has_device_id()) {
  acc_set_device_type(acc_device_host);
} else {
  ...
}

Reply:
What if `settings.has_device_id()` is not zero?

Reply:
`settings.has_device_id()` is `true` if the user explicitly asked for a device id in which case we do do not want to silently fall back onto the host.
If the runtime reports that they are at least one GPU we defer the selection to `Impl::get_gpu` in core so that alll backends behave the same.

Reply:
I realize now we should also NOT do the fallback if the user specified a GPU arch at configure time :/

Reply:
@dalg24 you are correct; to fall back to host, compiler option should be configured to support host too.

Reply:
`Implementations::get_gpu` will handle whether user-specified device id is valid or not, and thus we don't have to check the validity of the device id again here.